packages/frames.js/src/core/middleware/parseFramesMessage.test.ts

```diff
@@ -0,0 +1,111 @@
+    const context: FramesContext = {
+      request: new Request("https://example.com"),
+    } as any;
+
+    const mw = parseFramesMessage();
+    const response = redirect("http://test.com");
+    const next = jest.fn(() => Promise.resolve(response));
+
+    await expect(mw(context, next)).resolves.toMatchObject(response);
+    expect(next).toHaveBeenCalledWith();
+  });
+
+  it("moves to next middleware if request is POST but does not have a valid JSON body", async () => {
+    const context: FramesContext = {
+      request: new Request("https://example.com", {
+        method: "POST",
+        body: "invalid json",
+      }),
+    } as any;
+
+    const mw = parseFramesMessage();
+    const response = redirect("http://test.com");
+    const next = jest.fn(() => Promise.resolve(response));
+
+    await expect(mw(context, next)).resolves.toMatchObject(response);
+    expect(next).toHaveBeenCalledWith();
+  });
+
+  it("moves to next middleware if request is POST with valid JSON but invalid body shape", async () => {
+    const context: FramesContext = {
+      request: new Request("https://example.com", {
+        method: "POST",
+        body: JSON.stringify({}),
+      }),
+    } as any;
+
+    const mw = parseFramesMessage();
+    const response = redirect("http://test.com");
+    const next = jest.fn(() => Promise.resolve(response));
+
+    await expect(mw(context, next)).resolves.toMatchObject(response);
+    expect(next).toHaveBeenCalledWith();
+  });
+
+  it("parses frame message from request body and adds it to context", async () => {
+    const context: FramesContext = {
+      request: new Request("https://example.com", {
+        method: "POST",
+        body: JSON.stringify({
+          trustedData: {
+            messageBytes: Buffer.from(
+              Message.encode(
+                Message.create({
+                  data: {
+                    fid: 123,
+                    frameActionBody: {
+                      castId: {
+                        fid: 456,
+                      },
+                      address: Buffer.from([0x789]),
+                      buttonIndex: 1,
+                      inputText: Buffer.from("hello"),
+                      state: Buffer.from(JSON.stringify({ test: true })),
+                    },
+                  },
+                })
+              ).finish()
+            ).toString("hex"),
+          },
+          untrustedData: {},
+        }),
+      }),
+    } as any;
+
+    const mw = parseFramesMessage();
+    const next = jest.fn(() => Promise.resolve(new Response()));
+
+    await mw(context, next);
+
+    expect(next).toHaveBeenCalledWith({
+      message: {
+        buttonIndex: 1,
+        castId: {
+          fid: 456,
+          hash: "0x",
+        },
+        connectedAddress: "0x89",
+        inputText: "hello",
+        requesterFid: 123,
+        state: JSON.stringify({ test: true }),
```

Comment:
@stephancill if we treat state value as JSON I think we should automatically decode it right? So this should be just object, not string. What do you think?

Reply:
The middleware uses `getFrameMessage` under the hood and that function is not decoding the state as JSON. What should we do here?

Reply:
`getFrameMessage` cannot assume that the state is JSON unless it has some guarantee that it comes from a frames.js frame server, so I think this JSON deserialization should happen outside of `getFrameMessage`

Reply:
but we are implementing a library frames.js right? We don't aim to be compatible with states from other frames builders right?

Reply:
I added automatic decoding, if user uses frames.js to build frames, this works out of the box.

Reply:
sorry you're right. i was considering a case where `getFrameMessage` is used by something like `useFrame` or `FrameUI` but that is all client side. 

for us to be able to assume it's JSON we need to ensure that JSON ends up in frames.js meta tags AND that the user is using frames.js for generating meta tags. we obviously have control over how we generate the state meta tag and i am leaning of the assumption that the user is generating meta tags using frames.js if they are using `getFrameMessage`

Reply:
I added the JSON decoding to middleware so we still have a power to support anything in `state` since `getFrameMessage` treats it as string only. There is no change in that.